src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -317,73 +321,59 @@ static void fixPairOverlappingQualities(final PileupElement firstElement,
+    public ReadPileup getOverlappingFragmentFilteredPileup(boolean discardDiscordant, boolean baseQualNotMapQual) {
+        Map<String, PileupElement> filteredPileup = new HashMap<String, PileupElement>();
+
+        for (PileupElement p : this) {
+                String readName = p.getRead().getName();
 
-            for (PileupElement p : pileupElementTracker) {
-                String readName = p.getRead().getReadName();
+            // if we've never seen this read before, life is good
+            if (!filteredPileup.containsKey(readName)) {
+                filteredPileup.put(readName, p);
+            } else {
+                PileupElement existing = filteredPileup.get(readName);
 
-                // if we've never seen this read before, life is good
-                if (!filteredPileup.containsKey(readName)) {
-                    filteredPileup.put(readName, p);
+                // if the reads disagree at this position, throw them both out.  Otherwise
+                // keep the element with the higher quality score
+                if (discardDiscordant && existing.getBase() != p.getBase()) {
+                    filteredPileup.remove(readName);
                 } else {
-                    PileupElement existing = filteredPileup.get(readName);
-
-                    // if the reads disagree at this position, throw them both out.  Otherwise
-                    // keep the element with the higher quality score
-                    if (discardDiscordant && existing.getBase() != p.getBase()) {
-                        filteredPileup.remove(readName);
-                    } else {
-                        if (baseQualNotMapQual) {
-                            if (existing.getQual() < p.getQual())
-                                filteredPileup.put(readName, p);
-                        }
-                        else {
-                            if (existing.getMappingQual() < p.getMappingQual())
-                                filteredPileup.put(readName, p);
-                        }
+                    if (baseQualNotMapQual) {
+                        if (existing.getQual() < p.getQual())
+                            filteredPileup.put(readName, p);
+                    }
+                    else {
+                        if (existing.getMappingQual() < p.getMappingQual())
+                            filteredPileup.put(readName, p);
                     }
                 }
             }
+        }
 
-            UnifiedPileupElementTracker<PileupElement> filteredTracker = new UnifiedPileupElementTracker<PileupElement>();
-            for (PileupElement filteredElement : filteredPileup.values())
-                filteredTracker.add(filteredElement);
+        List<PileupElement> filteredPileupList = new ArrayList<PileupElement>();
+        for (PileupElement filteredElement : filteredPileup.values())
+            filteredPileupList.add(filteredElement);
```

Comment:
I think you can simplify this loop away with  `List<PileupElement> filteredPileupList = new ArrayList<>(filteredPileup.values())`